index.d.ts

```diff
@@ -0,0 +1,64 @@
+export const enum DataType {
+	string = 'string',
+}
+
+export const enum DataValueType {
+	string = 'string',
+}
+
+export interface DataValue {
+	type: DataValueType;
+	value: string;
```

Comment:
This actually shows the limitations of the “users can augment this with declaration merging” approach, because while declaration merging can add additional members, it can’t add additional union type branches to an individual member. So there’s no way to augment this type definition to make non-`string` values work at the moment.

I suspect the solution to this will involve something in the direction of 

```typescript
export interface DataValue {
	type: DataValueType;
	value: unknown;
}

export interface StringDataValue {
	type: DataValueTypeString;
	value: string;
}
```

but I’m not at all sure about the details yet, and would leave this for a future release.